File: Course II/С++/УП/Pract 5/Zad1.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int *a, i, k, n;

void initarray(){
    
    int one, two;
	cout<<"\nВведите n => "; cin>>n;
	a = new int [n];

    cout<<"Введите 1-й элемент массива => "; cin>>one;
    cout<<"Введите 2-й элемент массива => "; cin>>two;

	a[0]=one;
	a[1]=two;
	for(i=2;i<n;i++)
	    a[i]=(a[i-1]+a[i-2])*i;

}

void outarray(){

	cout<<"\nПреобразованный массив:\n[";
	for(i=0;i<n;i++)
    	cout<<a[i]<<" ";
	cout<<"]\n";

}

int main(){

    setlocale(LC_ALL,"rus");

    do
    {
     cout<<"\n\n1. Генерация массива\n2. Вывод массива\n0. Выход из программы\n=> ";
     cin>>k;
     switch (k)  
      {  
         case 1:  
            initarray();
            break;
          
         case 2:  
            outarray();
            break;

      }    

    } while (k!=0);

cout<<"\n";
return 0;

}

File: Course II/УП/Pract 4/Zad1.pas
program KOMUK;
uses crt;

var a: array of integer;
    k, n: integer;

procedure initarray();

  var one, two, i: integer;
  begin
    randomize;
    write('Введите кол-во элементов массива n => '); read(n);
    a:=new integer[n+1];
    write('Введите 1-й элемент массива => '); read(one);
    write('Введите 2-й элемент массива => '); read(two);
    
    a[1]:=one;
    a[2]:=two;

    for i:=3 to n do
      a[i]:=(a[i-1]+a[i-2])*i;

  end;

procedure outarray();
  var i: integer;
  begin
    writeln('Полученный массив:');
    write('[');
    for i:=1 to n do
      write(a[i],' ');
    write(']'); 
  readkey();
  end;
  
begin
 repeat
  clrscr;
  writeln('1. Генерация массива');
  writeln('2. Вывод массива');
  writeln('0. Выход из программы');
  write('=> '); readln(k);
  
  case k of
   1 : initarray();
   2 : outarray();
  end;
  until (k=0);
end.
